File: insert_BST.py
class Node:
    def __init__(self, val, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def insert_bst(bst: Node, val: int) -> Node:
    # WRITE YOUR BRILLIANT CODE HERE
    if not bst:
        return Node(val)
    if val < bst.val:
        bst.left = insert_bst(bst.left, val)
    elif val > bst.val:
        bst.right = insert_bst(bst.right, val)
    
    return bst

# this function build a tree from input
# learn more about how trees are encoded in https://algo.monster/problems/serializing_tree
def build_tree(nodes, f):
    val = next(nodes)
    if val == 'x': return None
    left = build_tree(nodes, f)
    right = build_tree(nodes, f)
    return Node(f(val), left, right)

def format_tree(node):
    if node is None:
        yield 'x'
        return
    yield str(node.val)
    yield from format_tree(node.left)
    yield from format_tree(node.right)

if __name__ == '__main__':
    bst = build_tree(iter(input().split()), int)
    val = int(input())
    res = insert_bst(bst, val)
    print(' '.join(format_tree(res)))

File: insert_bst.js
class Node {
    constructor(val, left = null, right = null) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

function insertBst(bst, val) {
    // WRITE YOUR BRILLIANT CODE HERE
    if (!bst) return new Node(val);
    if (bst.val > val) bst.left = insertBst(bst.left, val);
    else if (bst.val < val) bst.right = insertBst(bst.right, val);
    return bst;
}

// solution
function insertBst(bst, val) {
    if (bst == null) return new Node(val);
    if (bst.val < val) {
        bst.right = insertBst(bst.right, val);
    } else if (bst.val > val) {
        bst.left = insertBst(bst.left, val);
    }
    return bst;
}